File: backend-spring/src/main/java/com/sourcelens/module/issue/entity/IssueDecomposition.java
package com.sourcelens.module.issue.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@TableName("issue_decompositions")
public class IssueDecomposition {

    @TableId(type = IdType.AUTO)
    private Long id;

    private Long projectId;

    private Long scanTaskId;

    private String title;

    private String description;

    private String businessContext;

    private String priority;

    /** JSON 数组 */
    private String relatedModules;

    /** PENDING / PROCESSING / COMPLETED / FAILED */
    private String status;

    private String understanding;

    private String impactModules;

    private String impactApis;

    private String impactDb;

    private String risks;

    private String dependencies;

    private String acceptance;

    private String suggestedBranch;

    private String suggestedCommit;

    private String outputJson;

    private String errorMessage;

    private Long createdBy;

    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createdAt;

    @TableField(fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updatedAt;

    @TableLogic
    private Boolean deleted;
}

File: web-console/src/api/issueDecomposition.ts
import client from './client'
import type { Result } from './client'

export interface IssueDecomposition {
  id: number
  projectId: number
  scanTaskId: number | null
  title: string
  description: string
  businessContext: string | null
  priority: string
  relatedModules: string | null
  status: string
  understanding: string | null
  impactModules: string | null
  impactApis: string | null
  impactDb: string | null
  risks: string | null
  dependencies: string | null
  acceptance: string | null
  suggestedBranch: string | null
  suggestedCommit: string | null
  outputJson: string | null
  errorMessage: string | null
  createdBy: number
  createdAt: string
  updatedAt: string
}

export interface IssueTask {
  id: number
  decompositionId: number
  taskOrder: number
  category: string
  title: string
  description: string | null
  impactFiles: string | null
  riskLevel: string | null
  testSuggestions: string | null
  estimatedHours: number | null
  status: string
  createdAt: string
}

export interface PageResult<T> {
  items: T[]
  page: number
  pageSize: number
  total: number
}

export const issueApi = {
  create: (data: {
    projectId: number
    scanTaskId?: number
    title: string
    description: string
    businessContext?: string
    priority?: string
    relatedModules?: string
  }) => client.post<Result<IssueDecomposition>>('/issue-decompositions', data),

  listByProject: (projectId: number, page = 1, pageSize = 20, status?: string) => {
    const params = new URLSearchParams({ page: String(page), pageSize: String(pageSize) })
    if (status) params.set('status', status)
    return client.get<Result<PageResult<IssueDecomposition>>>(`/projects/${projectId}/issue-decompositions?${params}`)
  },

  detail: (id: number) =>
    client.get<Result<IssueDecomposition>>(`/issue-decompositions/${id}`),

  listTasks: (id: number) =>
    client.get<Result<IssueTask[]>>(`/issue-decompositions/${id}/tasks`),

  updateTaskStatus: (taskId: number, status: string) =>
    client.patch<Result<IssueTask>>(`/issue-tasks/${taskId}?status=${status}`),

  exportMarkdown: (id: number) =>
    client.get<Result<string>>(`/issue-decompositions/${id}/export/markdown`),
}